### PR TITLE
ES upgraded to 5.5.1

### PR DESCRIPTION
**Pull request by ashish-khatkar**

### CODE REVIEW

**Comment by shivam-rai** [comment]
1. Take backup of the currently deployed branch, and push it to Github, just in case it's needed.
2. There's a lot of commented out and unformatted code lying around. Please fix that. PRs should be raised with the expectation of merging the code, as is, to production/master. So, keep cleaning up code before we raise PRs.
3. Even if it means we'll need more time, let's keep the UTs in mind. Changes to 5.5.1 shouldn't be breaking existing UTs and if it needs additional coverage, let's add UTs. It'll help pitch our improvements to external teams, who might want to use our version of Foxtrot.

These are just some minor suggestions. We'll resume the review and merge, once you've covered them.

**Comment by ashish-khatkar** [comment]
1. Done. 
2. The commit related to the changes wasn't pushed.
3. No additional UTs required as I just upgraded the ES API.

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java [hunk 1]
```diff
@@ -110,8 +111,8 @@ public void save(String table, Document document) throws FoxtrotException {
                     .setType(ElasticsearchUtils.DOCUMENT_TYPE_NAME)
                     .setId(translatedDocument.getId())
                     .setTimestamp(Long.toString(timestamp))
-                    .setSource(convert(translatedDocument))
-                    .setConsistencyLevel(WriteConsistencyLevel.QUORUM)
```
Comment (shivam-rai) [inline foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java:114]: Why is this getting removed ?

Comment (shivam-rai) [reply foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java:114]: Doc for reference: https://www.elastic.co/guide/en/elasticsearch/reference/5.5/breaking_50_java_api_changes.html

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java [hunk 2]
```diff
@@ -183,15 +184,19 @@ public Document get(String table, String id) throws FoxtrotException {
...
+        /**
+         * Need to see what function should be used to replace setNoFields
+         * setNoFields
+         */
         SearchResponse searchResponse = connection.getClient()
                 .prepareSearch(ElasticsearchUtils.getIndices(table))
                 .setTypes(ElasticsearchUtils.DOCUMENT_TYPE_NAME)
                 .setQuery(boolQuery().filter(termQuery(ElasticsearchUtils.DOCUMENT_META_ID_FIELD_NAME, id)))
-                .setNoFields()
+//                .setNoFields()
```
Comment (shivam-rai) [inline foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java]: Reference: https://stackoverflow.com/questions/42789560/elasticsearch-upgrade-from-1-to-5-searchrequestbuilder-setnofields

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java [hunk 3]
```diff
@@ -183,15 +184,19 @@ public Document get(String table, String id) throws FoxtrotException {
         }
         fxTable = tableMetadataManager.get(table);
         String lookupKey;
+        /**
```
Comment (shivam-rai) [inline foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java]: Remove if not needed.

File: foxtrot-core/src/test/java/com/flipkart/foxtrot/core/MockElasticsearchServer.java [hunk 4]
```diff
@@ -36,17 +36,14 @@
...
-    public MockElasticsearchServer(String directory) {
+    public MockElasticsearchServer(String directory) throws NodeValidationException {
         this.DATA_DIRECTORY = UUID.randomUUID().toString() + "/" + directory;
         Settings settings = Settings.builder()
                 .put("http.enabled", "false")
                 .put("path.home", "target/" + DATA_DIRECTORY)
                 .build();
 
-        node = nodeBuilder()
-                .local(true)
```
Comment (shivam-rai) [inline foxtrot-core/src/test/java/com/flipkart/foxtrot/core/MockElasticsearchServer.java]: Just check once.

File: foxtrot-core/src/test/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStoreTest.java [hunk 5]
```diff
@@ -95,7 +94,9 @@ public void testSaveSingleRawKeyVersion1() throws Exception {
                 .prepareGet(ElasticsearchUtils.getCurrentIndex(TestUtils.TEST_TABLE_NAME, originalDocument.getTimestamp()),
                         ElasticsearchUtils.DOCUMENT_TYPE_NAME,
                         originalDocument.getId())
-                .setFields("_timestamp").execute().actionGet();
+                .setStoredFields("_timestamp")
```
Comment (shivam-rai) [inline foxtrot-core/src/test/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStoreTest.java]: [IMPORTANT] Check the sanity of these UTs. 
We need to check if there's any regression on the _timestamp field being removed.

**Comment by shivam-rai** [comment]
UTs failing
--------------------------------------------------  
testResolveNonExistentAction(com.flipkart.foxtrot.core.querystore.QueryExecutorTest)

Tests run: 317, Failures: 0, Errors: 252, Skipped: 3

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Foxtrot ............................................ SUCCESS [01:23 min]
[INFO] foxtrot-common ..................................... SUCCESS [  8.123 s]
[INFO] foxtrot-core ....................................... FAILURE [ 52.393 s]
[INFO] foxtrot-sql ........................................ SKIPPED
[INFO] foxtrot-server ..................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:24 min
[INFO] Finished at: 2017-12-11T16:04:15+05:30
[INFO] Final Memory: 72M/1477M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test (default-test) on project foxtrot-core: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/shivam.rai/code/foxtrot/foxtrot-core/target/surefire-reports for the individual test results.

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java (hunk 3 above)
Comment (shivam-rai) [reply foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java]: Kar de ye.

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java (hunk 2 above)
Comment (shivam-rai) [reply foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java]: Kar de ye.

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java (hunk 3 above)
Comment (ashish-khatkar) [reply foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java]: Arre ye kar diya. Ek jagah rha gya tha vo next PR main ho jayega remove.

File: foxtrot-core/src/test/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStoreTest.java [hunk 6]
```diff
@@ -393,14 +390,16 @@ public void testGetFieldMappings() throws FoxtrotException, InterruptedException
...
+        mappings.add(new FieldTypeMapping("word", FieldType.KEYWORD));
+        mappings.add(new FieldTypeMapping("data.data", FieldType.KEYWORD));
+        mappings.add(new FieldTypeMapping("header.hello", FieldType.KEYWORD));
         mappings.add(new FieldTypeMapping("head.hello", FieldType.LONG));
+        mappings.add(new FieldTypeMapping("timestamp", FieldType.DATE));
 
         TableFieldMapping tableFieldMapping = new TableFieldMapping(TestUtils.TEST_TABLE_NAME, mappings);
         TableFieldMapping responseMapping = queryStore.getFieldMappings(TestUtils.TEST_TABLE_NAME);
 
+        System.out.println(responseMapping.toString());
```
Comment (shivam-rai) [inline foxtrot-core/src/test/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStoreTest.java]: Remove please.

Comment (ashish-khatkar) [reply foxtrot-core/src/test/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStoreTest.java]: Not mine. Didn't see this. Will remove in next PR.

**Pull request merged**